src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -156,13 +156,13 @@ private IEnumerable<ConstructorSignature> BuildPrimaryConstructors(IReadOnlyList
 
             if (Fields.CredentialFields.Count == 0)
             {
-                yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalToRequired).ToArray());
+                yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalToRequired).OrderBy(parameter => parameter.Name != "endpoint").ToArray());
             }
             else
             {
                 foreach (var credentialField in Fields.CredentialFields)
                 {
-                    yield return CreatePrimaryConstructor(requiredParameters.Append(CreateCredentialParameter(credentialField!.Type)).Concat(optionalToRequired).ToArray());
+                    yield return CreatePrimaryConstructor(requiredParameters.Append(CreateCredentialParameter(credentialField!.Type)).Concat(optionalToRequired).OrderBy(parameter => parameter.Name != "endpoint").ToArray());
```

Comment:
this reads so much like a "patch" to our code.
Could we come up with a solution that does not look like a patch?

I heard the root cause of this is that in swagger, the endpoint is always optional. Is it possible we could make it required in the `CodeModelConverter` or anywhere else and then we do not need to do anything here and fix the issue?

Reply:
hello @ArcturusZhang The root cause is the endpoint is optional or it has default value. We also need to honor the default value and use it to call the secondary constructor in primary constructor. In that way, the endpoint parameter should maintain optional with default value. We cannot just convert it as required, and ignore the default value.